src/sentry/api/issue_search.py

```diff
@@ -289,7 +272,54 @@ def convert_search_filter(
             raise InvalidSearchQuery(
                 f"Aggregate filters ({search_filter.key.name}) are not supported in issue searches."
             )
+
         return search_filter
 
+    def expand_substatus_query_values(
+        search_filters: list[SearchFilter], org: Organization
+    ) -> list[SearchFilter]:
+        first_status_incl = None
+        first_status_excl = None
+        includes_status_filter = False
+        includes_substatus_filter = False
+        for search_filter in search_filters:
+            if search_filter.key.name == "substatus":
+                if not features.has("organizations:issue-states", org):
+                    raise InvalidSearchQuery(
+                        "The substatus filter is not supported for this organization"
+                    )
+
+                converted = convert_search_filter(search_filter, org)
+                new_value = converted.value.raw_value
+                status = GROUP_SUBSTATUS_TO_STATUS_MAP.get(
+                    new_value[0] if isinstance(new_value, list) else new_value
+                )
+                if first_status_incl is None and converted.operator in EQUALITY_OPERATORS:
+                    first_status_incl = SearchFilter(
+                        key=SearchKey(name="status"), operator="IN", value=SearchValue([status])
+                    )
+
+                if first_status_excl is None and converted.operator in INEQUALITY_OPERATORS:
+                    first_status_excl = SearchFilter(
+                        key=SearchKey(name="status"), operator="NOT IN", value=SearchValue([status])
+                    )
+
+                includes_substatus_filter = True
+
+            if search_filter.key.name == "status":
+                includes_status_filter = True
+
+        if includes_status_filter:
+            return search_filters
+
+        if includes_substatus_filter:
+            assert first_status_incl is not None or first_status_excl is not None
+            return search_filters + [first_status_incl or first_status_excl]
```

Comment:
There's some subtlety here where we prioritize the left-most search terms over the right ones. We also prioritize inclusive filters over exclusive filters. Calling it out here since I can imagine weird scenarios where the user intended to search for one thing but got some weird results.